Automattic/wp-calypso
# Social Previews | Hide Mastodon post preview when no connection

For Mastodon you see the "Your Post" section even if there is no account connected. This PR fixes that.

## Proposed Changes

* Hide Mastodon post preview when there is no connection
* Display link preview as an anonymous user even when the account is connected.

## Testing Instructions

- Checkout fire up this PR or use the Live link below
- Goto `/marketing/connections/:site`
- Ensure that there is no Mastodon connection
- Goto `/posts/:site` 
- Click on the 3 dots for any post and Click "Share"
- Click on "Preview"
- Select Mastodon preview
- Confirm that you only see the link preview and not the post preview
- Now goto `/marketing/connections/:site` again and connect a Mastodon account
- Reload /posts/:site`
- Click on the 3 dots for any post and Click "Share"
- Click on "Preview"
- Select Mastodon preview
- Confirm that you see both the post and link preview
- Confirm that the link preview is shown as an anonymous user

<table>
	<thead>
		<tr>
			<th>
				BEFORE
			</th>
			<th>
				AFTER
			</th>
		</tr>
	</thead>
	<tbody>
		<tr>
			<th colspan="2">
				When NO Mastodon connection exists
			</th>
		</tr>
		<tr>
			<td>
				<img width="1453" alt="Screenshot 2023-09-13 at 11 34 15 AM" src="https://github.com/Automattic/wp-calypso/assets/18226415/bbd69abb-5450-46b1-b0a0-239f953c649d">
			</td>
			<td>
				<img width="1453" alt="Screenshot 2023-09-13 at 11 33 49 AM" src="https://github.com/Automattic/wp-calypso/assets/18226415/feb6a7cc-0a09-4d12-b41c-6d53dd793471">
			</td>
		</tr>
		<tr>
			<th colspan="2">
				When Mastodon connection exists
			</th>
		</tr>
		<tr>
			<td>
				<img width="1450" alt="Screenshot 2023-09-13 at 11 32 21 AM" src="https://github.com/Automattic/wp-calypso/assets/18226415/faf61ad3-2264-4f14-895c-38bfd4606245">
			</td>
			<td>
				<img width="1449" alt="Screenshot 2023-09-13 at 11 33 03 AM" src="https://github.com/Automattic/wp-calypso/assets/18226415/2fc0c994-11e7-4dd5-a91a-9a40552fef74">
			</td>
		</tr>
	</tbody>

</table>

## Pre-merge Checklist
